### PR TITLE
Buildsystem ported to meson and added dark theme

### PR DESCRIPTION
**Pull request by Mau-G**
Most notable changes being upstreamed:

- Dark theme
- Build system change to Meson. Build system is noticeably faster (i.e. by the time Autotools finishes configuring, Meson has already built and installed)

Following previous feedback, everything is in there except:

- ~~Greybird-bright xfce-notify-4.0 (not needed as already handled by meson)~~
- Greybird ubiquity panel theme ( we no longer display the panel)

### CODE REVIEW

**Comment by bluesabre** [comment]
Thanks for this. I've done a bit of testing and everything seems to work, but there are a few adjustments and notes.

- The Greybird xfce4-notifyd theme should actually default to the dark theme, with Greybird-bright installed separately.
- The Greybird plank theme should be installed to share/themes/Greybird/plank. This enables automatically loading when Plank uses the "GTK+" theme. The same theme can also be installed in Greybird-dark.
- Greybird-Accessibility, Greybird-Compact, and Greybird-Dark should be Greybird-accessibility, Greybird-compact, and Greybird-dark respectively. This matches the current theme as well as upstream Adwaita.

File: MAINTAINING.md [hunk 1]
```diff
@@ -0,0 +1,6 @@
+# Maintainer's Guide to pulling commits from Greybird's upstream, Greybird
```
Comment (bluesabre) [inline MAINTAINING.md]: This file should not be needed, since Greybird is the upstream.

File: README.md [hunk 2]
```diff
@@ -30,23 +30,23 @@ The Gtk+3 theme uses the builtin engine and consequently has no dependencies.
 ### Build dependencies ###
 Debian or Ubuntu:
 
-`sudo apt install autoconf libgdk-pixbuf2.0-dev libglib2.0-bin librsvg2-dev ruby-sass`sassc
+`sudo apt install autoconf libgdk-pixbuf2.0-dev libglib2.0-bin librsvg2-dev ruby-sass sassc`
```
Comment (bluesabre) [inline README.md]: Debian/Ubuntu installations should also install meson

File: README.md [hunk 3]
```diff
@@ -30,23 +30,23 @@ The Gtk+3 theme uses the builtin engine and consequently has no dependencies.
...
 
 Fedora:
 
 `dnf install gdk-pixbuf2-devel librsvg2-devel rubygem-sass`
```
Comment (bluesabre) [inline README.md]: Fedora installations should also install meson

**Comment by Mau-G** [comment]
>     * The Greybird xfce4-notifyd theme should actually default to the dark theme, with Greybird-bright installed separately.

Now it's Greybird and Greybird-Dark which is consistent with the rest. Wouldn't it be simpler to make the notification Greybird-Dark default with Greybird  and Greybird-Dark themes? That's how we resolved it in openSUSE.

> 
>     * The Greybird plank theme should be installed to share/themes/Greybird/plank. This enables automatically loading when Plank uses the "GTK+" theme. The same theme can also be installed in Greybird-dark.

gotcha

>     * Greybird-Accessibility, Greybird-Compact, and Greybird-Dark should be Greybird-accessibility, Greybird-compact, and Greybird-dark respectively. This matches the current theme as well as upstream Adwaita.
> 

> Additionally, when installing I see several instances of this warning. Not sure if it's a problem.
> 
> `Installing /home/sean/Projects/Code/shimmer/Greybird/dark/xfwm4/top-right-inactive.xpm to /usr/local/share/themes/Greybird-Dark/xfwm4`
> `Warning: trying to copy a symlink that points to a file. This will copy the file, but this will be changed in a future version of Meson to copy the symlink as is. Please update your build definitions so that it will not break when the change happens.`



> Thanks for this. I've done a bit of testing and everything seems to work, but there are a few adjustments and notes.
> 
>     * The Greybird xfce4-notifyd theme should actually default to the dark theme, with Greybird-bright installed separately.
> 
>     * The Greybird plank theme should be installed to share/themes/Greybird/plank. This enables automatically loading when Plank uses the "GTK+" theme. The same theme can also be installed in Greybird-dark.
> 
Now it's Greybird and Greybird-Dark which is consistent with the rest. Wouldn't it be simpler to make the notification Greybird-Dark default with Greybird  and Greybird-Dark themes? That's how we resolved it in openSUSE.

>     * Greybird-Accessibility, Greybird-Compact, and Greybird-Dark should be Greybird-accessibility, Greybird-compact, and Greybird-dark respectively. This matches the current theme as well as upstream Adwaita.
> 
> 
> Additionally, when installing I see several instances of this warning. Not sure if it's a problem.
> 
> `Installing /home/sean/Projects/Code/shimmer/Greybird/dark/xfwm4/top-right-inactive.xpm to /usr/local/share/themes/Greybird-Dark/xfwm4`
> `Warning: trying to copy a symlink that points to a file. This will copy the file, but this will be changed in a future version of Meson to copy the symlink as is. Please update your build definitions so that it will not break when the change happens.`

**Comment by Mau-G** [comment]
> 
>     * The Greybird xfce4-notifyd theme should actually default to the dark theme, with Greybird-bright installed separately.

Now it's Greybird and Greybird-Dark which is consistent with the rest. Wouldn't it be better and simpler to make the notification Greybird-Dark default with Greybird  and Greybird-Dark themes? That's how we resolved it in openSUSE and would keep the same formula as everywhere else.

>     * The Greybird plank theme should be installed to share/themes/Greybird/plank. This enables automatically loading when Plank uses the "GTK+" theme. The same theme can also be installed in Greybird-dark.
> 

ok

>     * Greybird-Accessibility, Greybird-Compact, and Greybird-Dark should be Greybird-accessibility, Greybird-compact, and Greybird-dark respectively. This matches the current theme as well as upstream Adwaita.
> 
ok

> Additionally, when installing I see several instances of this warning. Not sure if it's a problem.
> 
> `Installing /home/sean/Projects/Code/shimmer/Greybird/dark/xfwm4/top-right-inactive.xpm to /usr/local/share/themes/Greybird-Dark/xfwm4`
> `Warning: trying to copy a symlink that points to a file. This will copy the file, but this will be changed in a future version of Meson to copy the symlink as is. Please update your build definitions so that it will not break when the change happens.`

I think we can ignore it but let me ask @appadeia

File: dark/meson.build [hunk 4]
```diff
@@ -1,7 +1,7 @@
-theme_path = join_paths(get_option('datadir'), 'themes/Greybird-Dark')
-a11y_theme_path = join_paths(get_option('datadir'), 'themes/Greybird-Dark-Accessibility')
-compact_theme_path = join_paths(get_option('datadir'), 'themes/Greybird-Dark-Compact')
-bright_theme_path = join_paths(get_option('datadir'), 'themes/Greybird-Dark-Bright')
+theme_path = join_paths(get_option('datadir'), 'themes/Greybird-dark')
+a11y_theme_path = join_paths(get_option('datadir'), 'themes/Greybird-Dark-accessibility')
+compact_theme_path = join_paths(get_option('datadir'), 'themes/Greybird-dark-compact')
+bright_theme_path = join_paths(get_option('datadir'), 'themes/Greybird-dark-bright')
```
Comment (Mau-G) [inline dark/meson.build]: eh this is kinda useless and should be removed

File: dark/meson.build [hunk 5]
```diff
@@ -1,7 +1,7 @@
-theme_path = join_paths(get_option('datadir'), 'themes/Greybird-Dark')
-a11y_theme_path = join_paths(get_option('datadir'), 'themes/Greybird-Dark-Accessibility')
-compact_theme_path = join_paths(get_option('datadir'), 'themes/Greybird-Dark-Compact')
-bright_theme_path = join_paths(get_option('datadir'), 'themes/Greybird-Dark-Bright')
+theme_path = join_paths(get_option('datadir'), 'themes/Greybird-dark')
+a11y_theme_path = join_paths(get_option('datadir'), 'themes/Greybird-Dark-accessibility')
```
Comment (bluesabre) [inline dark/meson.build]: This should probably be "Greybird-dark-accessibility"

**Comment by Mau-G** [comment]
> > Now it's Greybird and Greybird-Dark which is consistent with the rest. Wouldn't it be better and simpler to make the notification Greybird-Dark default with Greybird and Greybird-Dark themes? That's how we resolved it in openSUSE and would keep the same formula as everywhere else.
> 
> Normally it would, but Greybird is a "darkish" theme, defined by the light interface with dark panels and notifications. There's not a way to transition existing configurations to a different theme, such as longtime Xubuntu and Debian Greybird users. Currently the Greybird notification theme is dark, with Greybird-bright being the light option.

Ok, understood. Should not be trouble to fix it and make it like that

File: dark/meson.build (hunk 5 above)
Comment (Mau-G) [reply dark/meson.build]: fixed

**Pull request merged**